File: java/src/test/java/problems1_100/JumpGameTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class JumpGameTest extends SampledTest {
    @Test
    public void test() {
        for (JumpGame solution : JumpGame.values()) {
            for (JumpGameSample sample : getSamples(JumpGameSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.canJump(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/JumpGameTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class JumpGameTest : SampledTest() {
    @Test
    fun test() =
        JumpGame.entries.forEach { solution ->
            getSamples<Array<JumpGameSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.canJump(input.nums))
                    .isEqualTo(output)
            }
        }
}
